projects/angular/src/utils/focus-trap/focus-trap.spec.ts

```diff
@@ -0,0 +1,434 @@
+      expect(focusTrap.focusTrapElement.getAttribute('tabindex')).not.toBe('-1');
+    });
+
+    it('should add to focus trap attr on root element (html) to prevent scrolling', () => {
+      expect(document.documentElement.getAttribute(CDS_FOCUS_TRAP_DOCUMENT_ATTR)).toBe('');
+    });
+
+    it('should set itself to current on FocusTrapTracker service', () => {
+      expect(FocusTrapTrackerService.getCurrent()).toEqual(focusTrap.focusTrapElement as any);
+    });
+
+    it('should be focused', async () => {
+      // focus is not immediately set due to safari issue
+      await sleep(23);
+      expect(document.activeElement).toEqual(focusTrap.focusTrapElement, 'focus is set asynchronously after creation');
+    });
+
+    it('should be set to active', () => {
+      expect(focusTrap.active).toBe(true);
+    });
+
+    it('should throw an error if enabledFocusTrap is called again', () => {
+      const secondCall = () => focusTrap.enableFocusTrap();
+      expect(secondCall).toThrow();
+    });
+  });
+
+  describe('removeFocusTrap()', () => {
+    let focusTrap: FocusTrapComponent;
+    let fixture: ComponentFixture<TestFocusTrapComponent>;
+    let wrapperComponent: ComponentRef<TestFocusTrapComponent>;
+    let previousFocusedElement: HTMLElement;
+
+    beforeEach(async () => {
+      TestBed.configureTestingModule({
+        imports: [CommonModule],
+        declarations: [TestFocusTrapComponent, FocusTrapComponent],
+      });
+
+      fixture = TestBed.createComponent(TestFocusTrapComponent);
+      wrapperComponent = fixture.componentRef;
+      fixture.detectChanges();
+      focusTrap = wrapperComponent.instance.focusTrapComponent;
+      fixture.nativeElement.querySelector('button').focus();
+      previousFocusedElement = document.activeElement as HTMLElement;
+    });
+
+    afterEach(() => {
+      fixture.destroy();
+    });
+
+    it('should remove rebound elements', () => {
+      focusTrap.removeFocusTrap();
+      expect(document.querySelectorAll('.offscreen-focus-rebounder').length).toBe(0);
+    });
+
+    it('should remove layout attribute to prevent scrolling on body', () => {
+      expect(document.body.getAttribute('cds-layout')).toBeNull();
+    });
+
+    it('should not be set as current on FocusTrapTracker', () => {
+      expect(FocusTrapTrackerService.getCurrent()).not.toEqual(previousFocusedElement as any);
+    });
+
+    it('should not be set to active', () => {
+      expect(focusTrap.active).toBe(false);
+    });
+
+    it('should restore previous focus', () => {
+      focusTrap.removeFocusTrap();
+      expect(document.activeElement).toEqual(previousFocusedElement);
+    });
+  });
+});
+
+/**
+ * For some reason locally it runs ok, on the CI it says it doesn't.
+ *
+ * Have to revisit this but at this moment we don't use nested focus traps in our components, so...
+ * that should be okay for now.
+ */
+xdescribe('Nested FocusTraps: ', () => {
```

Comment:
Have to debug this one why it doesn't work on GitHub CI. Made a demo for it to test it manually, works fine but in order to not postpone this any more I'll would be happy to get this merged and revisit this in near future.

Reply:
In `@cds/core` we don't nest focus-traps, we _layer_ them so that only the top-most focus-trap is active. The approach simplifies all this. I'm not sure nested focus traps would be considered accessible.